src/util/detect-bugs.js

```diff
@@ -1,23 +1,32 @@
-import TRANSITIONRUN_EVENT_LOOP_BUG from "./detect-bugs/transitionrun-loop.js";
-import UNREGISTERED_TRANSITION_BUG from "./detect-bugs/unregistered-transition.js";
+import detectTransitionRunLoopBug from "./detect-bugs/transitionrun-loop.js";
+import detectUnregisteredTransitionBug from "./detect-bugs/unregistered-transition.js";
 
-/**
- * Data structure for all detected bugs.
- * All bugs start off as true, and once their promises resolve, that is replaced with the actual value
- */
-export const bugs = {
-	TRANSITIONRUN_EVENT_LOOP: true,
-	UNREGISTERED_TRANSITION: true,
-};
+// Cache for detection results
+let transitionRunLoopPromise = null;
+let unregisteredTransitionPromise = null;
 
-TRANSITIONRUN_EVENT_LOOP_BUG.then(value => {
-	bugs.TRANSITIONRUN_EVENT_LOOP = value;
-});
+const bugs = {
+	/**
+	 * Check if the browser is affected by the Safari transitionrun loop bug
+	 * @returns {Promise<boolean>}
+	 */
+	async transitionRunLoop() {
```

Comment:
Since this is an object, it can have accessors. That's a wonderful opportunity: it means we can define an API that _looks_ like regular properties, but runs the tests the first time the properties are called. Then, calling code doesn't even need to know about the functions.

Reply:
Interesting, but is it weird if the accessor is async?

Reply:
Or rather, I mean if the accessor returns a promise (which it has to). Maybe I'm misunderstanding what you're suggesting, but basically we could do this:

```diff
-	async transitionRunLoop() {
+	get transitionRunLoop() {
		if (!transitionRunLoopPromise) {
			transitionRunLoopPromise = detectTransitionRunLoopBug();
		}
		return transitionRunLoopPromise;
	},
```

But then the consumer still needs to do e.g. `if (await bugs.transitionRunLoop)` and not just `if (bugs.transitionRunLoop)`, which would feel a lot more natural and makes me think a function is better / clearer.

Reply:
I guess it didn't really have to return a promise. "Fixed" in https://github.com/LeaVerou/style-observer/commit/25d479c0ac47440c297aded338637a47d64062aa.